eucadeploy/plugins/debugger/verify_networking.py

```diff
@@ -0,0 +1,360 @@
+                      + ' with Object Storage Gateway ' + osg + ' on TCP port 8773') 
+            with hide('everything'):
+                iperf_result = self.execute_iperf_on_hosts(iperf_cmd,
+                                                           components)    
+
+            for component in components:
+                flag = False
+                for output in iperf_result[component].strip().split('\n'):
+                    if re.search('Connection refused', output):
+                        flag = True
+
+                if flag:
+                    self.failure(component + ':Eucalyptus Component was'
+                                 + ' not able to connect to Object Storage '
+                                 + 'Controller ' + osg + ' on TCP port 8773')
+                else:
+                    self.success(component + ':Eucalyptus Component'
+                                 + ' successfully connected to Object Storage '
+                                 + 'Controller ' + osg + ' on TCP port 8773')
+        
+    def _verify_java_db_comms(self, roles):
+        """
+        Verify Eucalyptus Java components can communicate to 
+        the database running on the Cloud Controller on 
+        TCP port 8777
+
+        :param roles: set of roles (cloud components) for a given 
+                      Eucalyptus cloud
+        """
+        java_components = roles['user-facing']
+
+        euca_java_roles = ['storage-controller']
+        if roles['walrus']:
+            euca_java_roles.append('walrus')
+
+        for component in euca_java_roles:
+            java_components = java_components.union(roles[component])
+
+        for host in roles['clc']:
+            iperf_cmd = 'iperf -c ' + host + ' -T 32 -t 5 -i 1 -p 8777'
+            self.info('Verifying Eucalyptus Java Components'
+                      + ' are able to communicate'
+                      + ' with Cloud Controller ' + host + ' on TCP port 8777') 
+            with hide('everything'):
+                iperf_result = self.execute_iperf_on_hosts(iperf_cmd,
+                                                           java_components)    
+
+            for component in java_components:
+                flag = False
+                for output in iperf_result[component].strip().split('\n'):
+                    if re.search('Connection refused', output):
+                        flag = True
+
+                if flag:
+                    self.failure(component + ':Eucalyptus Java Component was'
+                                 + ' not able to connect to Cloud '
+                                 + 'Controller ' + host + ' on TCP port 8777')
+                else:
+                    self.success(component + ':Eucalyptus Java Component'
+                                 + ' successfully connected to Cloud '
+                                 + 'Controller ' + host + ' on TCP port 8777')
+            
+
+    def _verify_clc_cc_comms(self, roles):
```

Comment:
Can we refactor these methods so that they reuse the same logic? From what I can tell the only real difference is the role and port.